bin_sort/BinSort.java

```diff
@@ -0,0 +1,75 @@
+import java.util.Random;
+
+private class Node {
+	double data;
+	Node next;

```

Comment:
@Hsankesara: Please mention these variables as `public`, `private` or `protected` based on the need.